pkg/apis/kubeone/validation/validation.go

```diff
@@ -174,11 +174,14 @@ func ValidateAPIEndpoint(a kubeoneapi.APIEndpoint, fldPath *field.Path) field.Er
 }
 
 // ValidateCloudProviderSpec validates the CloudProviderSpec structure
-func ValidateCloudProviderSpec(p kubeoneapi.CloudProviderSpec, fldPath *field.Path) field.ErrorList {
+func ValidateCloudProviderSpec(p kubeoneapi.CloudProviderSpec, networkConfig kubeoneapi.ClusterNetworkConfig, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
 	providerFound := false
 	if p.AWS != nil {
+		if networkConfig.IPFamily.IsDualstack() && len(p.CloudConfig) == 0 {
```

Comment:
Do we need cloud-config for IPv6-only clusters?

Reply:
No. It worked without any special config when I hacked it together.